util/auth.go

```diff
@@ -50,3 +52,17 @@ func GetTokenHash(token string) string {
 	// needs to be URL-encodable to send revoke token requests by hash
 	return b64.RawURLEncoding.EncodeToString(tokenHashBytes[:])
 }
+
+func WithUser(f func(echo.Context, *User) error) func(echo.Context) error {
```

Comment:
Copy-paste out of `/api/v1/handlers.go`. In the interest of keeping V1 untouched, I left the old one there. Should we make the refactor on V1, or leave as-is?

Reply:
you can do the refactor on V1 to prevent duplicating code here. if we need to change the middleware in the future and only apply it to newer api versions then we can make a separate func in this util and only apply it to the newer api versions.